paimon-core/src/main/java/org/apache/paimon/disk/InMemoryBuffer.java

```diff
@@ -68,7 +72,8 @@ private void tryInitialize() {
     @Override
     public void reset() {
         if (this.isInitialized) {
-            this.currentDataBufferOffset = 0;
+            // we need to set -1 here to avoid been preempt
+            this.currentDataBufferOffset = -1;
```

Comment:
I don't know why to set to -1.

I think we should modify `MemoryPoolFactory.preemptMemory`, if (memoryOccupancy > 0) assign to max.